include/dxc/Test/HlslTestUtils.h

```diff
@@ -517,6 +518,44 @@ inline bool CompareHalfEpsilon(const uint16_t &fsrc, const uint16_t &fref, float
   return std::abs(src_f32-ref_f32) < epsilon;
 }
 
+
+inline void ReplaceDisassemblyTextWithoutRegex(std::vector<std::string> &pLookFors,
+                            std::vector<std::string> &pReplacements,
+                            std::string &disassembly) {
+  for (unsigned i = 0; i < pLookFors.size(); ++i) {
+    LPCSTR pLookFor = pLookFors[i].data();
+    bool bOptional = false;
+    if (pLookFor[0] == '?') {
+      bOptional = true;
+      pLookFor++;
+    }
+    LPCSTR pReplacement = pReplacements[i].data();
+    if (pLookFor && *pLookFor) {
+      
+      bool found = false;
+      size_t pos = 0;
+      size_t lookForLen = strlen(pLookFor);
+      size_t replaceLen = strlen(pReplacement);
```

Comment:
There's actually no need to capture LPCSTR and size from `pReplacements[i]`, since the code could just use the object and size directly: `disassembly.replace(pos, lookForLen, pReplacements[i]);` ... `pos += pReplacements[i].size()`.